kube-core/src/request.rs

```diff
@@ -210,6 +210,24 @@ impl Request {
         req.body(vec![]).map_err(Error::BuildRequest)
     }
 
+    /// Create an instance of the subresource
+    pub fn create_subresource(
+        &self,
+        subresource_name: &str,
+        name: &str,
+        pp: &PostParams,
+        data: Vec<u8>,
+    ) -> Result<http::Request<Vec<u8>>, Error> {
+        let target = format!("{}/{}/{}?", self.url_path, name, subresource_name);
+        let mut qp = form_urlencoded::Serializer::new(target);
+        if pp.dry_run {
+            qp.append_pair("dryRun", "All");
+        }
+        let urlstr = qp.finish();
+        let req = http::Request::post(urlstr).header(http::header::CONTENT_TYPE, JSON_MIME);
+        req.body(data).map_err(Error::BuildRequest)
+    }
+
```

Comment:
this should have a basic test that at least the url is sensible, something like:

```rust
#[cfg(test)]
mod test {
    use crate::{request::Request, resource::Resource};
    use k8s_openapi::api::core::v1::ServiceAccount;

    #[test]
    fn create_subresource() {
        let url = ServiceAccount::url_path(&(), Some("ns"));
        let pp = PostParams::default();
        let data = vec![];
        let req = Request::new(url).create_subresource("token", "mysvc", &pp, data).unwrap();
        assert_eq!(req.uri(), "/api/v1/namespaces/ns/serviceaccounts/mysvc/token");
    }
}
```